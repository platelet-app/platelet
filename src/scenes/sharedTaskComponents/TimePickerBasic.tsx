import React from "react";
import TextField from "@mui/material/TextField";
import Autocomplete from "@mui/material/Autocomplete";
import { styled } from "@mui/styles";

type TimePickerBasicProps = {
    value: string;
    onChange: (value: string) => void;
    isValid?: boolean;
    showOnlyTodayTimes?: boolean;
    startValue?: string;
    showPlusOneDay?: boolean;
    label?: string;
};

const CustomTextField = styled(TextField)({
    "& .MuiFormHelperText-root": {
        position: "absolute",
        top: "100%",
    },
});

const TimePickerBasic: React.FC<TimePickerBasicProps> = ({
    value,
    onChange,
    isValid,
    showOnlyTodayTimes = false,
    startValue,
    showPlusOneDay = false,
    label = "Time",
}) => {
    const timeOptions = generateTimeOptions(startValue, showOnlyTodayTimes);

    const handleInputChange = (_: any, newInputValue: string) => {
        // Regular expression to allow only numbers and colons, with a maximum length of 5
        const validInput = newInputValue.replace(/[^0-9:]/g, "").slice(0, 5);
        onChange(validInput);
    };

    let helperText = "";
    if (showPlusOneDay) {
        helperText = "+1 day";
    } else if (!isValid) {
        helperText = "Invalid time";
    }

    return (
        <Autocomplete
            fullWidth
            freeSolo
            id="time-picker-basic"
            options={timeOptions}
            getOptionLabel={(option) => option}
            value={value}
            inputValue={value}
            onInputChange={handleInputChange}
            onChange={(_, newValue) => {
                if (typeof newValue === "string") {
                    onChange(newValue);
                }
            }}
            renderInput={(params) => (
                <CustomTextField
                    {...params}
                    helperText={helperText}
                    error={!isValid}
                    label={label}
                    variant="outlined"
                    sx={{
                        helperText: {
                            "& .MuiFormHelperText-root": {
                                height: "0",
                                marginTop: "0",
                            },
                        },
                    }}
                />
            )}
        />
    );
};

function generateTimeOptions(startTime?: string, onlyToday?: boolean) {
    const options = [];
    let currentHour = 0;
    if (startTime) {
        currentHour = parseInt(startTime.split(":")[0]);
    } else if (onlyToday) {
        currentHour = new Date().getHours();
    }
    for (let hour = currentHour; hour < 24; hour++) {
        for (let minute = 0; minute < 60; minute += 30) {
            const timeString = `${hour.toString().padStart(2, "0")}:${minute
                .toString()
                .padStart(2, "0")}`;
            options.push(timeString);
        }
    }
    if (onlyToday) return options;
    for (let hour = 0; hour < 24 - (24 - currentHour); hour++) {
        for (let minute = 0; minute < 60; minute += 30) {
            const timeString = `${hour.toString().padStart(2, "0")}:${minute
                .toString()
                .padStart(2, "0")}`;
            options.push(timeString);
        }
    }
    console.log("options", options);
    return options;
}

export default TimePickerBasic;
