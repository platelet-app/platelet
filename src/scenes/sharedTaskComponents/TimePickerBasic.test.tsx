import * as React from "react";
import userEvent from "@testing-library/user-event";
import { render, screen, waitFor } from "../../test-utils";
import TimePickerBasic from "./TimePickerBasic";

describe("TimePickerBasic", () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date("2021-01-01T12:00:00Z"));
    });
    afterEach(() => {
        jest.useRealTimers();
    });
    test("choose a time from given options", async () => {
        const onChange = jest.fn();
        render(<TimePickerBasic value="" onChange={onChange} />);
        const input = screen.getByRole("textbox");
        userEvent.click(input);
        expect(screen.getAllByRole("option")).toMatchSnapshot();
        userEvent.click(await screen.findByRole("option", { name: "12:00" }));
        expect(onChange).toHaveBeenCalledWith("12:00");
    });
    test("write a time", async () => {
        const onChange = jest.fn();
        render(<TimePickerBasic value="" onChange={onChange} />);
        const input = screen.getByRole("combobox");
        userEvent.type(input, "12:34");
        expect(onChange.mock.calls).toMatchInlineSnapshot(`
            Array [
              Array [
                "1",
              ],
              Array [
                "2",
              ],
              Array [
                ":",
              ],
              Array [
                "3",
              ],
              Array [
                "4",
              ],
            ]
        `);
    });
    test("show only today's times", async () => {
        const onChange = jest.fn();
        render(
            <TimePickerBasic showOnlyTodayTimes value="" onChange={onChange} />
        );
        userEvent.click(screen.getByRole("textbox"));
        await waitFor(() => {
            expect(
                screen.getByRole("option", { name: "12:30" })
            ).toBeInTheDocument();
        });
        expect(screen.queryByRole("option", { name: "11:30" })).toBeNull();
        expect(screen.queryByRole("option", { name: "00:00" })).toBeNull();
        expect(
            screen.getByRole("option", { name: "23:30" })
        ).toBeInTheDocument();
        expect(screen.getAllByRole("option")).toMatchSnapshot();
    });
    test("show if invalid time", async () => {
        const onChange = jest.fn();
        render(
            <TimePickerBasic value="" isValid={false} onChange={onChange} />
        );
        expect(screen.getByText("Invalid time")).toBeInTheDocument();
    });
    test("show +1 day", async () => {
        const onChange = jest.fn();
        render(<TimePickerBasic value="" showPlusOneDay onChange={onChange} />);
        expect(screen.getByText("+1 day")).toBeInTheDocument();
    });
    test("set the start time", async () => {
        const onChange = jest.fn();
        render(
            <TimePickerBasic value="" startValue="12:00" onChange={onChange} />
        );
        userEvent.click(screen.getByRole("textbox"));
        await waitFor(() => {
            expect(
                screen.getByRole("option", { name: "12:00" })
            ).toBeInTheDocument();
        });
        const options = screen.getAllByRole("option");
        expect(options).toMatchSnapshot();
    });
});
