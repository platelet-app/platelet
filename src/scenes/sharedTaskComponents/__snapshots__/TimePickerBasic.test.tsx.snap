// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePickerBasic choose a time from given options 1`] = `
Array [
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="0"
    id="time-picker-basic-option-0"
    role="option"
    tabindex="-1"
  >
    00:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="1"
    id="time-picker-basic-option-1"
    role="option"
    tabindex="-1"
  >
    00:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="2"
    id="time-picker-basic-option-2"
    role="option"
    tabindex="-1"
  >
    01:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="3"
    id="time-picker-basic-option-3"
    role="option"
    tabindex="-1"
  >
    01:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="4"
    id="time-picker-basic-option-4"
    role="option"
    tabindex="-1"
  >
    02:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="5"
    id="time-picker-basic-option-5"
    role="option"
    tabindex="-1"
  >
    02:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="6"
    id="time-picker-basic-option-6"
    role="option"
    tabindex="-1"
  >
    03:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="7"
    id="time-picker-basic-option-7"
    role="option"
    tabindex="-1"
  >
    03:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="8"
    id="time-picker-basic-option-8"
    role="option"
    tabindex="-1"
  >
    04:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="9"
    id="time-picker-basic-option-9"
    role="option"
    tabindex="-1"
  >
    04:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="10"
    id="time-picker-basic-option-10"
    role="option"
    tabindex="-1"
  >
    05:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="11"
    id="time-picker-basic-option-11"
    role="option"
    tabindex="-1"
  >
    05:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="12"
    id="time-picker-basic-option-12"
    role="option"
    tabindex="-1"
  >
    06:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="13"
    id="time-picker-basic-option-13"
    role="option"
    tabindex="-1"
  >
    06:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="14"
    id="time-picker-basic-option-14"
    role="option"
    tabindex="-1"
  >
    07:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="15"
    id="time-picker-basic-option-15"
    role="option"
    tabindex="-1"
  >
    07:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="16"
    id="time-picker-basic-option-16"
    role="option"
    tabindex="-1"
  >
    08:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="17"
    id="time-picker-basic-option-17"
    role="option"
    tabindex="-1"
  >
    08:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="18"
    id="time-picker-basic-option-18"
    role="option"
    tabindex="-1"
  >
    09:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="19"
    id="time-picker-basic-option-19"
    role="option"
    tabindex="-1"
  >
    09:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="20"
    id="time-picker-basic-option-20"
    role="option"
    tabindex="-1"
  >
    10:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="21"
    id="time-picker-basic-option-21"
    role="option"
    tabindex="-1"
  >
    10:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="22"
    id="time-picker-basic-option-22"
    role="option"
    tabindex="-1"
  >
    11:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="23"
    id="time-picker-basic-option-23"
    role="option"
    tabindex="-1"
  >
    11:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="24"
    id="time-picker-basic-option-24"
    role="option"
    tabindex="-1"
  >
    12:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="25"
    id="time-picker-basic-option-25"
    role="option"
    tabindex="-1"
  >
    12:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="26"
    id="time-picker-basic-option-26"
    role="option"
    tabindex="-1"
  >
    13:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="27"
    id="time-picker-basic-option-27"
    role="option"
    tabindex="-1"
  >
    13:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="28"
    id="time-picker-basic-option-28"
    role="option"
    tabindex="-1"
  >
    14:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="29"
    id="time-picker-basic-option-29"
    role="option"
    tabindex="-1"
  >
    14:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="30"
    id="time-picker-basic-option-30"
    role="option"
    tabindex="-1"
  >
    15:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="31"
    id="time-picker-basic-option-31"
    role="option"
    tabindex="-1"
  >
    15:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="32"
    id="time-picker-basic-option-32"
    role="option"
    tabindex="-1"
  >
    16:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="33"
    id="time-picker-basic-option-33"
    role="option"
    tabindex="-1"
  >
    16:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="34"
    id="time-picker-basic-option-34"
    role="option"
    tabindex="-1"
  >
    17:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="35"
    id="time-picker-basic-option-35"
    role="option"
    tabindex="-1"
  >
    17:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="36"
    id="time-picker-basic-option-36"
    role="option"
    tabindex="-1"
  >
    18:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="37"
    id="time-picker-basic-option-37"
    role="option"
    tabindex="-1"
  >
    18:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="38"
    id="time-picker-basic-option-38"
    role="option"
    tabindex="-1"
  >
    19:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="39"
    id="time-picker-basic-option-39"
    role="option"
    tabindex="-1"
  >
    19:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="40"
    id="time-picker-basic-option-40"
    role="option"
    tabindex="-1"
  >
    20:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="41"
    id="time-picker-basic-option-41"
    role="option"
    tabindex="-1"
  >
    20:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="42"
    id="time-picker-basic-option-42"
    role="option"
    tabindex="-1"
  >
    21:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="43"
    id="time-picker-basic-option-43"
    role="option"
    tabindex="-1"
  >
    21:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="44"
    id="time-picker-basic-option-44"
    role="option"
    tabindex="-1"
  >
    22:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="45"
    id="time-picker-basic-option-45"
    role="option"
    tabindex="-1"
  >
    22:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="46"
    id="time-picker-basic-option-46"
    role="option"
    tabindex="-1"
  >
    23:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="47"
    id="time-picker-basic-option-47"
    role="option"
    tabindex="-1"
  >
    23:30
  </li>,
]
`;

exports[`TimePickerBasic set the start time 1`] = `
Array [
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="0"
    id="time-picker-basic-option-0"
    role="option"
    tabindex="-1"
  >
    12:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="1"
    id="time-picker-basic-option-1"
    role="option"
    tabindex="-1"
  >
    12:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="2"
    id="time-picker-basic-option-2"
    role="option"
    tabindex="-1"
  >
    13:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="3"
    id="time-picker-basic-option-3"
    role="option"
    tabindex="-1"
  >
    13:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="4"
    id="time-picker-basic-option-4"
    role="option"
    tabindex="-1"
  >
    14:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="5"
    id="time-picker-basic-option-5"
    role="option"
    tabindex="-1"
  >
    14:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="6"
    id="time-picker-basic-option-6"
    role="option"
    tabindex="-1"
  >
    15:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="7"
    id="time-picker-basic-option-7"
    role="option"
    tabindex="-1"
  >
    15:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="8"
    id="time-picker-basic-option-8"
    role="option"
    tabindex="-1"
  >
    16:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="9"
    id="time-picker-basic-option-9"
    role="option"
    tabindex="-1"
  >
    16:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="10"
    id="time-picker-basic-option-10"
    role="option"
    tabindex="-1"
  >
    17:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="11"
    id="time-picker-basic-option-11"
    role="option"
    tabindex="-1"
  >
    17:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="12"
    id="time-picker-basic-option-12"
    role="option"
    tabindex="-1"
  >
    18:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="13"
    id="time-picker-basic-option-13"
    role="option"
    tabindex="-1"
  >
    18:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="14"
    id="time-picker-basic-option-14"
    role="option"
    tabindex="-1"
  >
    19:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="15"
    id="time-picker-basic-option-15"
    role="option"
    tabindex="-1"
  >
    19:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="16"
    id="time-picker-basic-option-16"
    role="option"
    tabindex="-1"
  >
    20:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="17"
    id="time-picker-basic-option-17"
    role="option"
    tabindex="-1"
  >
    20:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="18"
    id="time-picker-basic-option-18"
    role="option"
    tabindex="-1"
  >
    21:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="19"
    id="time-picker-basic-option-19"
    role="option"
    tabindex="-1"
  >
    21:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="20"
    id="time-picker-basic-option-20"
    role="option"
    tabindex="-1"
  >
    22:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="21"
    id="time-picker-basic-option-21"
    role="option"
    tabindex="-1"
  >
    22:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="22"
    id="time-picker-basic-option-22"
    role="option"
    tabindex="-1"
  >
    23:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="23"
    id="time-picker-basic-option-23"
    role="option"
    tabindex="-1"
  >
    23:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="24"
    id="time-picker-basic-option-24"
    role="option"
    tabindex="-1"
  >
    00:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="25"
    id="time-picker-basic-option-25"
    role="option"
    tabindex="-1"
  >
    00:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="26"
    id="time-picker-basic-option-26"
    role="option"
    tabindex="-1"
  >
    01:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="27"
    id="time-picker-basic-option-27"
    role="option"
    tabindex="-1"
  >
    01:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="28"
    id="time-picker-basic-option-28"
    role="option"
    tabindex="-1"
  >
    02:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="29"
    id="time-picker-basic-option-29"
    role="option"
    tabindex="-1"
  >
    02:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="30"
    id="time-picker-basic-option-30"
    role="option"
    tabindex="-1"
  >
    03:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="31"
    id="time-picker-basic-option-31"
    role="option"
    tabindex="-1"
  >
    03:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="32"
    id="time-picker-basic-option-32"
    role="option"
    tabindex="-1"
  >
    04:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="33"
    id="time-picker-basic-option-33"
    role="option"
    tabindex="-1"
  >
    04:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="34"
    id="time-picker-basic-option-34"
    role="option"
    tabindex="-1"
  >
    05:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="35"
    id="time-picker-basic-option-35"
    role="option"
    tabindex="-1"
  >
    05:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="36"
    id="time-picker-basic-option-36"
    role="option"
    tabindex="-1"
  >
    06:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="37"
    id="time-picker-basic-option-37"
    role="option"
    tabindex="-1"
  >
    06:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="38"
    id="time-picker-basic-option-38"
    role="option"
    tabindex="-1"
  >
    07:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="39"
    id="time-picker-basic-option-39"
    role="option"
    tabindex="-1"
  >
    07:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="40"
    id="time-picker-basic-option-40"
    role="option"
    tabindex="-1"
  >
    08:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="41"
    id="time-picker-basic-option-41"
    role="option"
    tabindex="-1"
  >
    08:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="42"
    id="time-picker-basic-option-42"
    role="option"
    tabindex="-1"
  >
    09:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="43"
    id="time-picker-basic-option-43"
    role="option"
    tabindex="-1"
  >
    09:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="44"
    id="time-picker-basic-option-44"
    role="option"
    tabindex="-1"
  >
    10:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="45"
    id="time-picker-basic-option-45"
    role="option"
    tabindex="-1"
  >
    10:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="46"
    id="time-picker-basic-option-46"
    role="option"
    tabindex="-1"
  >
    11:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="47"
    id="time-picker-basic-option-47"
    role="option"
    tabindex="-1"
  >
    11:30
  </li>,
]
`;

exports[`TimePickerBasic show only today's times 1`] = `
Array [
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="0"
    id="time-picker-basic-option-0"
    role="option"
    tabindex="-1"
  >
    12:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="1"
    id="time-picker-basic-option-1"
    role="option"
    tabindex="-1"
  >
    12:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="2"
    id="time-picker-basic-option-2"
    role="option"
    tabindex="-1"
  >
    13:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="3"
    id="time-picker-basic-option-3"
    role="option"
    tabindex="-1"
  >
    13:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="4"
    id="time-picker-basic-option-4"
    role="option"
    tabindex="-1"
  >
    14:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="5"
    id="time-picker-basic-option-5"
    role="option"
    tabindex="-1"
  >
    14:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="6"
    id="time-picker-basic-option-6"
    role="option"
    tabindex="-1"
  >
    15:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="7"
    id="time-picker-basic-option-7"
    role="option"
    tabindex="-1"
  >
    15:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="8"
    id="time-picker-basic-option-8"
    role="option"
    tabindex="-1"
  >
    16:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="9"
    id="time-picker-basic-option-9"
    role="option"
    tabindex="-1"
  >
    16:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="10"
    id="time-picker-basic-option-10"
    role="option"
    tabindex="-1"
  >
    17:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="11"
    id="time-picker-basic-option-11"
    role="option"
    tabindex="-1"
  >
    17:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="12"
    id="time-picker-basic-option-12"
    role="option"
    tabindex="-1"
  >
    18:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="13"
    id="time-picker-basic-option-13"
    role="option"
    tabindex="-1"
  >
    18:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="14"
    id="time-picker-basic-option-14"
    role="option"
    tabindex="-1"
  >
    19:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="15"
    id="time-picker-basic-option-15"
    role="option"
    tabindex="-1"
  >
    19:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="16"
    id="time-picker-basic-option-16"
    role="option"
    tabindex="-1"
  >
    20:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="17"
    id="time-picker-basic-option-17"
    role="option"
    tabindex="-1"
  >
    20:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="18"
    id="time-picker-basic-option-18"
    role="option"
    tabindex="-1"
  >
    21:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="19"
    id="time-picker-basic-option-19"
    role="option"
    tabindex="-1"
  >
    21:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="20"
    id="time-picker-basic-option-20"
    role="option"
    tabindex="-1"
  >
    22:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="21"
    id="time-picker-basic-option-21"
    role="option"
    tabindex="-1"
  >
    22:30
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="22"
    id="time-picker-basic-option-22"
    role="option"
    tabindex="-1"
  >
    23:00
  </li>,
  <li
    aria-disabled="false"
    aria-selected="false"
    class="MuiAutocomplete-option"
    data-option-index="23"
    id="time-picker-basic-option-23"
    role="option"
    tabindex="-1"
  >
    23:30
  </li>,
]
`;
